### ISSUE TITLE
provide a constant for digest length size

### ISSUE THREAD

**Issue by cardoe** [RustCrypto/hashes#43]
Please provide a constant for digest length size from each hash type. e.g.

https://github.com/mitsuhiko/rust-sha1/blob/9c41f8eac17b64ef4a3d5d483dc658a8b8491a19/src/lib.rs#L24

**Comment by burdges** [comment]
Right now, the digest crate and these hashes use the generic_array crate because Rust only gained const generics recently.  I'd hope that once const generics stabilize the generic_array crate will be removed from all crates including this one, and doing so means adding a constant for the digest length.

**Comment by newpavlov** [comment]
On stabilization of const generics I'll definitely migrate to them, but right now as @burdges said we are tied to `generic-array` and `typenum`. If you want to retrieve digest output size at runtime, you can write the following code:
```Rust
use generic-array::typenum::Unsigned;
let n = Sha256::OutputSize::to_usize();
```
We could use associated constant in addition to `OutputSize`, but right now it's fairly useless and I don't see much benefit in introducing it.

Could you please provide use-case for which you need this constant? Otherwise I am inclined to close this issue.

**Comment by cardoe** [comment]
Plenty of file formats and other protocols internally include a hash in their data. They typically support multiple hashes in that case and have some form of identifier. When dealing with a binary format you decide which hash is in the data based on the identifier and then read `DIGEST_LENGTH` of bytes from the data. I honestly expected the `Digest` trait to provide a length method that was `#[inline]` that returned the size of the digest. The example provided above by @newpavlov is a bit clunky to say the least.

**Comment by newpavlov** [comment]
Well, I guess until const generics are introduced we probably could add `output_size` method to `FixedOutput` and `Digest` traits.

**Comment by newpavlov** [comment]
Fixed with `digest v0.8`. See `output_size` method on `Digest` trait.
